dart/lib/flat_buffers.dart

```diff
@@ -439,7 +443,12 @@ class Builder {
 
   /// Reset the builder and make it ready for filling a new buffer.
   void reset() {
-    _buf = new ByteData(initialSize);
+    if (_buf == null) {
+      _buf = _allocator.allocate(initialSize);
+      _allocator.clear(_buf, true);
+    } else {
+      _allocator.clear(_buf, false);
+    }
```

Comment:
Why not just make `allocate` (or some similarly named method) create the new cleared buffer?

This seems like it would potentially leave us with a buffer that was grown past the initial size after a reset.

Reply:
If you choose to reuse an existing builder, it could be beneficial (faster) to reuse the (allocated) buffer as well. This is also what FBB does for other languages, e.g.
* go builder Reset(): https://github.com/google/flatbuffers/blob/master/go/builder.go#L45
* c++ builder Clear(): https://github.com/google/flatbuffers/blob/master/include/flatbuffers/flatbuffers.h#L1224

Also, whether `_allocator.clear()` is necessary at all is a different topic. I don't know FB internals enough to know whether the unused memory must be zeroed out. It was, in the original allocator implementation implicitly (using `ByteData()`) but other allocators may have to do some work to clear it (as in the given `Pointer<Uint8>` example).